akka-http/src/main/scala/akka/http/javadsl/server/directives/BasicDirectives.scala

```diff
@@ -283,4 +285,26 @@ abstract class BasicDirectives {
    */
   def extractRequestEntity(inner: JFunction[RequestEntity, Route]): Route = extractEntity(inner)
 
+  /**
+   * WARNING: This will read the entire request entity into memory regardless of size and effectively disable streaming.
+   *
+   * Extracts the [[akka.http.javadsl.model.HttpEntity.Strict]] from the [[akka.http.javadsl.server.RequestContext]].
```

Comment:
`Converts the HttpEntity from the [[akka.http.javadsl.server.RequestContext]] into an [[akka.http.javadsl.model.HttpEntity.Strict]], or fails the route if unable to drain the entire request body within the timeout`